alvr/server_core/src/tracking/vmc.rs

```diff
@@ -0,0 +1,169 @@
+        (*BODY_RIGHT_FOOT_ID, "RightFoot"),
+        (*HEAD_ID, "Head"),
+    ])
+});
+
+static DEVICE_MOTIONS_ROTATION_MAP: Lazy<HashMap<u64, Quat>> = Lazy::new(|| {
+    HashMap::from([
+        (
+            *HAND_LEFT_ID,
+            Quat::from_xyzw(
+                -6.213430570750633e-08,
+                -1.7979416426202113e-07,
+                0.7071067411992601,
+                0.7071065897770331,
+            ),
+        ),
+        (
+            *HAND_RIGHT_ID,
+            Quat::from_xyzw(
+                0.3219087189747184,
+                0.7288832784221684,
+                0.3392694392278636,
+                -0.4999999512887856,
+            ),
+        ),
+        (
+            *BODY_CHEST_ID,
+            Quat::from_xyzw(
+                -0.48548752779498533,
+                0.5137675867233772,
+                -0.5131288074683715,
+                -0.4868713724975375,
+            ),
+        ),
+        (
+            *BODY_HIPS_ID,
+            Quat::from_xyzw(
+                -0.38920734358587283,
+                0.4731761921254787,
+                -0.25037835650145845,
+                -0.7496216673275858,
+            ),
+        ),
+        (
+            *BODY_LEFT_KNEE_ID,
+            Quat::from_xyzw(
+                0.42592041950463216,
+                0.5417784481730453,
+                0.3880448422573891,
+                -0.6119550893141193,
+            ),
+        ),
+        (
+            *BODY_LEFT_FOOT_ID,
+            Quat::from_xyzw(
+                -0.15405857492497116,
+                0.6901198926998461,
+                -2.2748115460768936e-08,
+                -0.7071068140641757,
+            ),
+        ),
+        (
+            *BODY_RIGHT_KNEE_ID,
+            Quat::from_xyzw(
+                -0.4650211913041333,
+                0.5058610693118244,
+                -0.6180250915435218,
+                -0.381974687482609,
+            ),
+        ),
+        (
+            *BODY_RIGHT_FOOT_ID,
+            Quat::from_xyzw(
+                0.6815561983146281,
+                -0.18836473838436454,
+                0.7071065841771154,
+                5.6213965690665724e-08,
+            ),
+        ),
+    ])
+});
+
+pub struct VMCSink {
+    socket: Option<UdpSocket>,
+}
+
+impl VMCSink {
+    pub fn new(config: VMCConfig) -> Result<Self> {
+        let socket = UdpSocket::bind("0.0.0.0:0")?;
+        socket.connect(format!("{}:{}", config.host, config.port))?;
+
+        Ok(Self {
+            socket: Some(socket),
+        })
+    }
+
+    fn send_osc_message(&self, path: &str, args: Vec<OscType>) {
+        if let Some(socket) = &self.socket {
+            socket
+                .send(
+                    &rosc::encoder::encode(&OscPacket::Message(OscMessage {
+                        addr: path.into(),
+                        args,
+                    }))
+                    .unwrap(),
+                )
+                .ok();
```

Comment:
Might want to log errors that occur here

Reply:
I based myself on the one implemented here https://github.com/alvr-org/ALVR/blob/master/alvr/server_core/src/tracking/body.rs#L48-L60
So if it's really needed, adding error logging there could also be helpful.

Reply:
Tbh it's probably fine without error logging, was just a hunch